src/components/Detailpage/Detailpage.js

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { ReactComponent as Gluten } from '../../images/icons/gluten.svg'
+import { ReactComponent as Soy } from '../../images/icons/soy.svg'
+import { ReactComponent as Arrow } from '../../images/icons/back-arrow.svg'
+import { useHistory, useParams } from 'react-router-dom'
+
+export default function Detailpage({ category }) {
+  let { id } = useParams()
+  useEffect(() => {
+    fetchDetail()
+  }, [])
+
+  const [detail, setDetail] = useState({
+    allergies: [],
+  })
+
+  const fetchDetail = async () => {
```

Comment:
Please make this a function with params `category` and `id` and extract it to `/services/getDetails.js`.